pkg/cri/server/image_pull.go

```diff
@@ -366,13 +369,24 @@ func (c *criService) registryHosts(auth *runtime.AuthConfig) docker.RegistryHost
 
 // defaultScheme returns the default scheme for a registry host.
 func defaultScheme(host string) string {
+	if isLocalHost(host) {
+		return "http"
+	}
+	return "https"
+}
+
+// isLocalHost checks if the registry host is local.
+func isLocalHost(host string) bool {
 	if h, _, err := net.SplitHostPort(host); err == nil {
 		host = h
 	}
-	if host == "localhost" || host == "127.0.0.1" || host == "::1" {
-		return "http"
+
+	if host == "localhost" {
+		host = "127.0.0.1"
```

Comment:
or ::1 ?

Reply:
not sure if ip.IsLoopback() will return true if not setup, such as if only setup for ipv6?

Reply:
what do you mean by "setup"? the implementation for `ip.IsLoopback()` handles both v4 and v6 IP addresses and returns true based on parsing 127.* for IPv4 and ::1 for IPv6.

Reply:
I suppose my question is.. is localhost for ipv4 guaranteed to be configured on all platforms where go resolves?

Reply:
I mean if SupportsIPv4() == false... we might not want to map localhost to 127.0.0.1 ..?

Reply:
I see ... the is loopback function is a simple check.. in this case always returns true.. so we could just return here on "localhost" with true..

Reply:
Done